#!/usr/bin/env python
PACKAGE = "rnw_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("desired_nutation", double_t, 0, "desired nutation degree", 30, 0, 90)
gen.add("tau", double_t, 0, "rocking angle for each step, along the contact normal vector", 30, 0, 60)
gen.add("rocking_max_vel", double_t, 0, "max_vel for rocking", 2, 0, 5)
gen.add("rocking_max_acc", double_t, 0, "max_acc for rocking", 2, 0, 5)
gen.add("ang_vel_threshold", double_t, 0, "min spin velocity when rocking", 0.1, 0, 1)
gen.add("waiting_ratio", double_t, 0, "waiting_ratio", 1, 0, 1)
gen.add("direct_control", bool_t, 0, "direct_control",  False)

exit(gen.generate(PACKAGE, "rnw_ros", "RNW"))
